contrib/pylightning/lightning/lightning.py

```diff
@@ -25,25 +25,20 @@ def _writeobj(sock, obj):
         s = json.dumps(obj)
         sock.sendall(bytearray(s, 'UTF-8'))
 
-    def _readobj(self, sock):
-        buff = b''
+    def _readobj(self, sock, buff=b''):
+        """Read a JSON object, starting with buff; returns object and any buffer left over"""
         while True:
-            try:
+            parts = buff.split(b'\n\n', 1)
+            if len(parts) == 1:
+                # Didn't read enough.
                 b = sock.recv(1024)
                 buff += b
                 if len(b) == 0:
-                    return {'error': 'Connection to RPC server lost.'}
-
-                if buff[-3:] != b' }\n':
-                    continue
-
-                # Convert late to UTF-8 so glyphs split across recvs do not
-                # impact us
-                objs, _ = self.decoder.raw_decode(buff.decode("UTF-8"))
-                return objs
-            except ValueError:
-                # Probably didn't read enough
-                pass
+                    return {'error': 'Connection to RPC server lost.'}, buff
+            else:
+                buff = parts[1]
+                obj, _ = self.decoder.raw_decode(parts[0].decode("UTF-8"))
```

Comment:
Nit: Being paranoid, you could still keep the try / `except ValueError` on `raw_decode` call to catch if there is ever un-decodable garbage in the buffer